src/boilerplate-card-editor.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  css,
+  CSSResult
+} from "lit-element";
+import { Switch } from "@material/mwc-switch";
+import { superstruct } from "superstruct";
+
+import { BoilerplateConfig, HomeAssistant, ValidHassDomEvent } from "./types";
+
+const cardConfigStruct = superstruct({
+  type: "string",
+  name: "string?",
+  show_error: "boolean?",
+  show_warning: "boolean?"
+});
+
+@customElement("boilerplate-card-editor")
+export class BoilerplateCardEditor extends LitElement {
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: BoilerplateConfig;
+
+  public setConfig(config: BoilerplateConfig): void {
+    config = cardConfigStruct(config);
+    this._config = config;
+  }
+
+  get _name(): string {
+    return this._config!.name || "";
+  }
+
+  get _show_error(): boolean {
+    return this._config!.show_error || false;
+  }
+
+  get _show_warning(): boolean {
+    return this._config!.show_warning || false;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this.hass) {
+      return html``;
+    }
+
+    return html`
+      <mwc-switch>sentiment_very_satisfied</mwc-switch>
+    `;
+
+    // return html`
+    //   <div class="card-config">
+    //     <paper-input
+    //       label="Name (Optional)"
+    //       .value="${this._name}"
+    //       .configValue="${' name'}"
+    //       @value-changed="${this._valueChanged}"
+    //     ></paper-input>
+    //     <div class="side-by-side">
+    //       <paper-toggle-button
+    //         ?checked="${this._show_error !== false}"
+    //         .configValue="${'show_error'}"
+    //         @change="${this._valueChanged}"
+    //         >Show Error?</paper-toggle-button
+    //       >
+    //       <paper-toggle-button
+    //         ?checked="${this._show_warning !== false}"
+    //         .configValue="${'show_warning'}"
+    //         @change="${this._valueChanged}"
+    //         >Show Warning?</paper-toggle-button
+    //       >
+    //     </div>
+    //   </div>
+    // `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      paper-toggle-button {
+        padding-top: 16px;
+      }
+      .side-by-side {
+        display: flex;
+      }
+      .side-by-side > * {
+        flex: 1;
+        padding-right: 4px;
+      }
+    `;
+  }
+
+  private _valueChanged(ev: Event): void {
+    if (!this._config || !this.hass) {
+      return;
+    }
+    const target = ev.target! as any;
+
+    if (
+      this[`_${target.configValue}`] === target.value ||
```

Comment:
I did this as well on the light-entity card at first but say you change the temperature to 100 then decide to change the color to . The color temperature isn't valid anymore but it's still at 100 so if I wanted to go back to the previous color temperature I want to send the same value it was previously. Something to think about at least.